### PR TITLE
PVC resize fixes

### PR DESCRIPTION
**Pull request by harsh-px**
- don't return just on triggering pvc resize. can lead to multiple resizes
- checks for existing pvc resize in progress to avoid stacking up multiple resizes
- return err if action has failed so that it would be retried
- add short name for autopilot rule crd

### CODE REVIEW

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -38,22 +54,76 @@ func (k *inst) ResizeVolume(volumeName string, resizePercentage int) error {
...
+
 	scaledValue := float64(current) * float64(resizePercentage) / 100
-	currentSize.Add(*resource.NewQuantity(int64(scaledValue), resource.DecimalSI))
-	logrus.Debugf("new PVC size = %v (%v)", currentSize.Value(), currentSize.Format)
-	pvc.Spec.Resources.Requests[v1.ResourceStorage] = currentSize
 
-	_, err = k8s.Instance().UpdatePersistentVolumeClaim(pvc)
+	newSize := currentSize.DeepCopy()
+	newSize.Add(*resource.NewQuantity(int64(scaledValue), resource.DecimalSI))
+	logrus.Debugf("new PVC size = %v (%v)", newSize.Value(), newSize.Format)
```
Comment (trierra) [inline drivers/scheduler/k8s/k8s.go]: maybe add PVC id/rule name to logs in case if multiple PVC in progress

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -38,22 +54,76 @@ func (k *inst) ResizeVolume(volumeName string, resizePercentage int) error {
...
 	}
 
-	currentSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+	specSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+	logrus.Debugf("PVC size in spec = %v %v.", specSize.Value(), specSize.Format)
 
-	logrus.Debugf("current PVC size = %v (%v)", currentSize.Value(), currentSize.Format)
+	currentSize := pvc.Status.Capacity[v1.ResourceStorage]
 	current := currentSize.Value()
+	logrus.Debugf("PVC actual size in status = %v %v.", currentSize.Value(), currentSize.Format)
```
Comment (trierra) [inline drivers/scheduler/k8s/k8s.go]: I'd add PVC id/rule name

File: drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (harsh-px) [reply drivers/scheduler/k8s/k8s.go]: done

Comment (harsh-px) [reply drivers/scheduler/k8s/k8s.go]: added pvc name, namespace and pv namea

File: drivers/scheduler/k8s/k8s.go (hunk 2 above)
Comment (harsh-px) [reply drivers/scheduler/k8s/k8s.go]: added pvc name, namespace and pv namea

Comment (harsh-px) [reply drivers/scheduler/k8s/k8s.go]: at the k8s layer, we don't have knowledge of rule names.

**Pull request merged**